### ISSUE TITLE
Is it possible to completely remove "From: HEADER?"

### ISSUE THREAD

**Issue by Quqas** [marlam/msmtp-mirror#69]
```
:~# msmtp --version
msmtp version 1.8.16
Platform: x86_64-pc-linux-gnu
```
Example:
`echo -e "Subject:Hi\n\nGoodbye" | msmtp -d root`
Options: in /etc/msmtprc
1)`#from good@ya.com`
msmtp: envelope-from address is missing
2)
```
set_from_header on
from good@ya.com
```

In the received mail From: good@ya.com
3)
 ```
set_from_header off
from good@ya.com
```

In received mail From: MAILER-DAEMON

those. if there is no From: in the body of the letter, then everything works.

but if there is From: wrong@xz.biz then we get
`smtpstatus=550 smtpmsg='550 5.7.0 Sender or From header address rejected: not owned by authorized user`

Of course the first option, the correct filling of From:
but I would like the second option, for example `set_from_header remove` (or I don’t understand how to implement this in the current version)

I want to use _msmtp**d**_ to forward emails from any address using a single account, but I can't configure sender address substitution.

1) `msmtpd --interface=192.168.25.88 --port=10025 --command='/bin/msmtp -f %F'`
=>
auth=on user=good@ya.com from=**_wrong_**@xz.biz recipients=test@test.ru smtpstatus=553 smtpmsg='553 5.7.1 Sender address rejected: not owned by auth user.

2)`msmtpd --interface=192.168.25.88 --port=10025 --command='/bin/msmtp'`
=>
auth=on user=good@ya.com from=**_good_**@ya.com recipients=test@test.ru smtpstatus=550 smtpmsg='550 5.7.0 Sender or From header address rejected: not owned by authorized user

translate.google.ru

**Comment by marlam** [comment]
It is now common that SMTP servers not just check the envelope-from address but also the From header for validity, which is a good thing. So you want both to be correct, and for that you use `from good@ya.com` and `set_from_header on`.

However, a program calling msmtp might still use the `-f` option to override the address you specified with the `from` command. To prevent that, use `allow_from_override off`. Note that `allow_from_override` is not yet available in a release version.

It is not possible with msmtp to remove a From header without replacing it, because then the mail would be invalid.

Does that answer your question?

**Comment by Quqas** [comment]
OK. Thanks. I'll wait for "allow_from_override" to be released

**Comment by marlam** [comment]
I just released version 1.8.20 with `allow_from_override`.
